lib/components/searchbox.js

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import Mousetrap from 'mousetrap';
+
+import Component from '../component';
+
+class Searchform extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchTerm: '',
+      searchError: false
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleSearchTermChange = this.handleSearchTermChange.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    // If a submit button was selected, return focus back to the input box
+    this.searchBox.focus();
+
+    const term = this.props.terms && this.props.terms.getActiveTerm();
+    if (term) {
+      const searchTerm = this.state.searchTerm;
+      const options = {
+        dir: this.state.dir
```

Comment:
dir is a poor name, do you mean directory or direction?  (I am going to guess directory every time, especially when working with a console application).  Avoid abbreviations, it's more important that your code be easily readable than easily typable.